cedar-policy-core/src/est/policy_set.rs

```diff
@@ -36,6 +35,67 @@ pub struct PolicySet {
     pub links: Vec<Link>,
 }
 
+impl PolicySet {
+    /// Get the static or template-linked policy with the given id.
+    /// Returns an `Option` rather than a `Result` because it is expected to be
+    /// used in cases where the policy set is guaranteed to be well-formed
+    /// (e.g., after successful conversion to an `ast::PolicySet`)
+    pub fn get_policy(&self, id: &PolicyID) -> Option<Policy> {
+        let maybe_static_policy = self
+            .static_policies
+            .iter()
+            .filter_map(|entry| {
+                if &entry.id == id {
+                    Some(entry.policy.clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>()
+            .first()
```

Comment:
```suggestion
            .next()
```

Reply:
Thanks for the review lol. Figured I was doing something weird here.